pkg/fqdn/dnsproxy/udp.go

```diff
@@ -55,37 +60,39 @@ type sessionUDP struct {
 	oob   []byte
 }
 
-var ciliumSessionUDPFactory = &sessionUDPFactory{}
```

Comment:
Removing this isn't great. We the options controlling this are parsed after init() or package variables would be created, though. Perhaps it is better to make this an init function like:
```
var sesssionFactoryOnce sync.Once

func makeCiliumSessionUDPFactory(ipv4, ipv6 bool) *ciliumSessionUDPFactory {
	sesssionFactoryOnce.Do(func() {
		ciliumSessionUDPFactory = &sessionUDPFactory{ipv4: ipv4, ipv6: ipv6}
	})
	return ciliumSessionUDPFactory
}
```

Reply:
I see no harm in deleting this. Simpler the way you changed it in this PR.